ziggurat-core-crawler/src/summary.rs

```diff
@@ -0,0 +1,70 @@
+use std::{cmp, collections::HashMap, fmt, fs, path::Path, time::Duration};
+
+use serde::Serialize;
+use spectre::graph::AGraph;
+
+#[derive(Default, Clone, Serialize)]
+pub struct NetworkSummary {
+    pub num_known_nodes: usize,
+    pub num_good_nodes: usize,
+    pub num_known_connections: usize,
+    pub num_versions: usize,
+    pub protocol_versions: HashMap<u32, usize>,
+    pub user_agents: HashMap<String, usize>,
+    pub crawler_runtime: Duration,
+    pub node_ips: Vec<String>,
```

Comment:
Just to think about if it shouldn't be Vec<IpAddr> - maybe it would be easier to handle when it's string.

Reply:
So far it's Vec of String in every project. Right now I'm just extracting common parts, we can think about refactoring later.